### ISSUE TITLE
Possible path collision when calculating route to an area we don't own any countries in

### ISSUE THREAD

**Issue by JohnMTorgerson** [JohnMTorgerson/Viking#3]
When Viking doesn't own any countries in an area, it calculates the cheapest route to that area from a country it does own, which of course means it's passing through countries not in the area it wants. If, on the same turn, the bot also wants to take over areas containing those countries it passes through on the way to the first area, the paths it calculates for the second area will not know about the path through it that was calculated for the first area, and the bot will get stuck in the attack phase when it hits the countries it has already taken over.

We have added logic to trim the colliding paths in the second area so that the bot will treat them as forks of the cheapest-route path that they would have collided with, but this does not solve all possible cases. In particular, if the paths intersect in an 'X' pattern, the newer path will be trimmed to fork from the older path, and the countries before the intersection will no longer be accounted for.

This scenario has proven to be relatively rare, but not so rare that it shouldn't be fixed eventually. And of course, even though the bot's attack plan will fail in this situation, it can fix it up fine in the next turn.

**Comment by JohnMTorgerson** [comment]
We've decided to address this issue by checking for the existence of the country in the <battlePlan> array list in most cases where we check a country to see if we own it. So when we're finding paths for an objective, any countries we're already planning on taking over in another objective will be treated as if we already own them, and the new paths will be calculated with that in mind, forking off of those countries as necessary.

In addition to solving the collision problem, this will have the added benefit of opening up new objectives for placeArmies() to choose from that would have been too expensive otherwise, which will now appear cheaper if we've already taken over some of the countries necessary to pursue it. For example, if we want to take over a certain continent, we might then decide to takeover a second continent behind the first, which we had to go through to get there. Taking the second continent from scratch might have been too expensive (after paying for the first objective), but now we can calculate the cost as though we've already covered part of the distance, which was already paid for. So we can essentially daisy-chain through objectives, which should be a big efficiency gain in some cases.

We haven't finished implementing this fix, but we started to do so in the last commit.
